wgpu-hal/src/vulkan/instance.rs

```diff
@@ -756,6 +756,15 @@ impl crate::Instance<super::Api> for super::Instance {
                 create_info = create_info.push_next(vk_create_info);
             }
 
+            let mut gpu_assisted_validation = vk::ValidationFeaturesEXT::builder()
+                .enabled_validation_features(&[vk::ValidationFeatureEnableEXT::GPU_ASSISTED]);
+            if desc
+                .flags
+                .intersects(wgt::InstanceFlags::GPU_BASED_VALIDATION)
+            {
+                create_info = create_info.push_next(&mut gpu_assisted_validation);
+            }
+
```

Comment:
I noticed that the `PRINTF_SHADER` validation feature conflicts with `GPU_ASSISTED` (viz., the one this PR exposes): https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/6ebabb71e297e0b8348ee4dd0633f2af3daaa771/docs/debug_printf.md#L174
 
I interpret this to mean that this change will conflict with #4297 (CC @exrook). We need to resolve this conflict with _some_ API design, but I have not yet determined a recommendation.

Reply:
☝🏻CC @cwfitzgerald, @teoxoy.

Reply:
Let's give priority to gpu based validation - if GBV is enabled, the printf feature isn't advertised.

Reply:
@exrook: ☝🏻Then it's a race to whoever gets merged last to implement this! 😈🏁🏃💨